lib/requester/requester.js

```diff
@@ -411,15 +411,21 @@ _.assign(Requester.prototype, /** @lends Requester.prototype */ {
         }
 
         return requests(request, requestOptions, onStart, function (err, res, resBody, debug) {
+            // prepare history from request debug data
+            var history = getExecutionHistory(debug),
+                responseTime,
+                response;
+
             if (err) {
                 // bubble up http errors
                 // @todo - Should we send an empty sdk Response here?
-                return onEnd(err);
+                //
+                // Also sending `history` object to enable consumers to
+                // reliably figure out if session was created, hence enabling them to
+                // know whether it was a Request Error or Response Error.
```

Comment:
These business logic details are not required to commented here.
This is just ensuring that the `history` object is sent even in case of error.